crates/consensus/src/transaction/deposit.rs

```diff
@@ -111,6 +111,47 @@ impl TxDeposit {
         mem::size_of::<bool>() + // is_system_transaction
         self.input.len() // input
     }
+
+    /// Get the transaction type
+    pub(crate) const fn tx_type(&self) -> OpTxType {
+        OpTxType::Deposit
+    }
+
+    /// Inner encoding function that is used for both rlp [`Encodable`] trait and for calculating
+    /// hash that for eip2718 does not require rlp header
+    pub fn encode_inner(&self, out: &mut dyn BufMut, with_header: bool) {
```

Comment:
let's use this here: https://github.com/alloy-rs/op-alloy/blob/6e19eb2bec8383e9d3de45fe0f4e528b12769986/crates/consensus/src/transaction/envelope.rs#L359

Reply:
done